performance-tests/test-app/element-rendering.js

```diff
@@ -0,0 +1,36 @@
+const { registerHtml, useStore } = require('../../src/tram-one')
+
+const html = registerHtml()
+
+/**
+ * This page has an input that changes the total number of elements on the page
+ */
+module.exports = () => {
+	const pageStore = useStore({ queue: '1000', elements: '1000', startTimer: 0, endTimer: 0, renders: 0 })
+
+	const updateCount = event => {
+		pageStore.queue = event.target.value
+	}
+
+	const render = () => {
+		pageStore.renders++
+		pageStore.startTimer = performance.now()
+		pageStore.elements = pageStore.queue
+		pageStore.endTimer = performance.now()
```

Comment:
these happen one at a time, which is why this whole performance test works. 

I wonder if this should be fixed though, so that all the state changes are queued and made at once, and then we rerender? I don't know if this would be technically challenging, or even possible 🤔